streaming/stream_transfer_task.cc

```diff
@@ -270,7 +229,7 @@ future<> stream_transfer_task::execute() {
             if (si->db.features().cluster_supports_stream_with_rpc_stream()) {
                 return send_mutation_fragments(std::move(si));
             } else {
-                return send_mutations(std::move(si));
+                throw std::runtime_error("cluster does not support STREAM_WITH_RPC_STREAM feature");
```

Comment:
Can we try to be a bit friendlier to the one odd user who decided to try their luck migration Scylla 2.0 to Scylla 4.2 ?
The message should maybe explain why the failure, that they should use an intermediate version, and preferably which version.

Reply:
I can expand the message, though I fear a user trying to add a 4.2 node to a  2.3 cluster isn't very good at reading messages.

Reply:
Let's act like God and show infinite patience by giving the user one more change.
And true to character, if they don't listen to this last advice we're justified in flooding their home and killing everyone inside